drivers/samsung/displays/mdc_protocol.cr

```diff
@@ -0,0 +1,406 @@
+require "driver/interface/powerable"
+require "driver/interface/muteable"
+require "driver/interface/switchable"
+
+module Samsung; end
+
+class Samsung::Displays::MDCProtocol < PlaceOS::Driver
+  include Interface::Powerable
+  include Interface::Muteable
+
+  enum Inputs
+    Vga           = 0x14 # pc in manual
+    Dvi           = 0x18
+    Dvi_Video     = 0x1F
+    Hdmi          = 0x21
+    Hdmi_Pc       = 0x22
+    Hdmi2         = 0x23
+    Hdmi2_Pc      = 0x24
+    Hdmi3         = 0x31
+    Hdmi3_Pc      = 0x32
+    Hdmi4         = 0x33
+    Hdmi4_Pc      = 0x34
+    Display_Port  = 0x25
+    Dtv           = 0x40
+    Media         = 0x60
+    Widi          = 0x61
+    Magic_Info    = 0x20
+    Whiteboard    = 0x64
+  end
+  include PlaceOS::Driver::Interface::InputSelection(Inputs)
+
+  # Discovery Information
+  tcp_port 1515
+  descriptive_name "Samsung MD, DM & QM Series LCD"
+  generic_name :Display
+
+  # Markdown description
+  description <<-DESC
+  For DM displays configure the following 1:
+
+  1. Network Standby = ON
+  2. Set Auto Standby = OFF
+  3. Set Eco Solution, Auto Off = OFF
+
+  Hard Power off displays each night and hard power ON in the morning.
+  DESC
+
+  default_settings({
+    display_id: 0,
+    rs232_control: false
+  })
+
+  @id : Int32 = 0
+  @rs232 : Bool = false
+  @blank : String?
+
+  # TODO: figure out how to define indicator \xAA
+  def init_tokenizer
+    @buffer = Tokenizer.new do |io|
+      bytes = io.peek
+      logger.debug { "Received: #{bytes}" }
+      # [header, command, id, data.size, [data], checksum]
+      bytes[3].to_i + 5
+    end
+  end
+
+  def on_load
+    transport.tokenizer = init_tokenizer
+    on_update
+
+    self[:volume_min] = 0
+    self[:volume_max] = 100
+    self[:previous_volume] = 50
```

Comment:
ah okay yeah that makes sense. I've removed volume_min and volume_max since they're not used anywhere as well https://github.com/PlaceOS/drivers/pull/19/commits/5dad71750b639eb3ef8cd05ebb7d260d9c4dd0a4